slick.groupitemmetadataprovider.js

```diff
@@ -42,8 +42,18 @@
       includeHeaderTotals: false
     };
 
-    options = $.extend(true, {}, _defaults, options);
+    var options = {};
 
+    setOptions(inputOptions);
+
+    function getOptions(){
+      return options;
+    }
+
+    function setOptions(inputOptions)
+    {
+      options = $.extend(true, {}, _defaults, options, inputOptions);
```

Comment:
considering what I wrote on the other extend that you removed, after putting it back in the `init()`, you should change this extend here with `options = $.extend({}, options, newOptions);` Same as what I did in the Grid Menu on this [line](https://github.com/6pac/SlickGrid/blob/master/controls/slick.gridmenu.js#L150), basically this should only change the options you are passing, nothing else and certainly not the defaults

Reply:
please check line 47 :) or the comment on your code review

Reply:
using `$.extend(true, {}, _defaults, ...)` ensures that the `_defaults` value is not changed (also show in following GIF.  
in fact, it doesn't change any of the objects other than the first (`target`) according to its [documentation](https://api.jquery.com/jquery.extend/)

`init` is not called on creation. as a result, the `options` will not be populated unless the plugin is initiated. the way I have called the function ensures the functionality is not changed in any way, as the execution path stays the same (extending the object using the `$.extend` function is done on virtually the same execution path location).  
I could easily change the execution path and call the `setOption` in `init` function, but cannot guaranty that the functionality doesn't break - in case `options` is used for any reason before plugin initialization.

![no-defaults-change](https://user-images.githubusercontent.com/2896170/75050962-4cb23e00-54cd-11ea-85a7-51bc41c9c087.gif)

Reply:
the defaults are to be used only if you don't have anything set in your options and should be used only on first file load not afterward. What is wrong with the original code `options = $.extend(true, {}, _defaults, options);`? Does that give you issues?

Reply:
well, the way I have defined the function, the `_defaults` is practically only going to be used on first initiation. true, it will be used every time, but then overwritten by the values already set to `options`.  
the problem would be that the same code then would practically be used twice: once during the construction, once during the `setOptions`

Reply:
The thing is that I would still prefer to see this `options = $.extend({}, options, newOptions);` in the `setOptions`, that is without the defaults

Reply:
so like I said I would prefer if you do the changes mentioned, but if you want to stick with your code, at least put it on the same line so it's more obvious that it's called and set, at first glance I didn't see it at all.

Reply:
I would work on it on Monday.